common/src/lib.rs

```diff
@@ -241,8 +241,15 @@ where
         context_switch_function_start_frame,
         context_switch_function_start_frame + 1,
     ) {
+        let page = Page::containing_address(VirtAddr::new(frame.start_address().as_u64()));
         match unsafe {
-            kernel_page_table.identity_map(frame, PageTableFlags::PRESENT, frame_allocator)
+            kernel_page_table.map_to_with_table_flags(
+                page,
+                frame,
+                PageTableFlags::PRESENT,
+                PageTableFlags::PRESENT | PageTableFlags::WRITABLE,
+                frame_allocator,
+            )
```

Comment:
Hey. I'm not sure you saw my latest commit. Github seems not to show the comments I added because they are before this code here. They are visible in the "Files changed" tab. 

Let me know if I should be more specific, but I personally prefer a shorter comment with the reason and a link to the issue over a longer explanation.

Reply:
Thanks for reminding me, I must have missed that notification. The comments look good.

Reply:
Seeing as I missed one of your requested changes, I can totally understand how that might happen :)